components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/dao/WorkflowDAO.java

```diff
@@ -137,6 +137,30 @@ public void removeWorkflow(String workflowId) throws InternalWorkflowException {
         }
     }
 
+    /**
+     * Remove all workflows of a given tenant id.
+     *
+     * @param tenantId Id of the tenant
+     * @throws InternalWorkflowException
+     */
+    public void removeWorkflowByTenantId(int tenantId) throws InternalWorkflowException {
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection(true);
+        PreparedStatement prepStmt = null;
+        String query = SQLConstants.DELETE_WORKFLOW_BY_TENANT_ID_QUERY;
+        try {
```

Comment:
use try with resource for the connection and the prepared statement. Fix in all the places.